test/N3Store-test.js

```diff
@@ -299,43 +301,119 @@ describe('Store', () => {
+      new Quad(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')), new NamedNode('p1'), new NamedNode('o1')),
+      new Quad(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')), new NamedNode('p2'), new NamedNode('o2')),
+      new Quad(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')), new NamedNode('p1'), new NamedNode('o2')),
+      new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2')),
+    ];
+    beforeAll(() => {
+      store = new Store(allQuads);
+    });
+
+    it('should start with the correct size', () => {
+      expect(store.size).toEqual(5);
     });
 
     it(
       'should return the removed quads',
-      forResultStream(shouldIncludeAll, () => { return store.removeMatches(null, 'p2', 'o2'); },
-        [termToId(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1'))), 'p2', 'o2'])
+      () => arrayifyStream(store.removeMatches(null, 'p2', 'o2')).then(quads => {
+        expect(quads.length).toBe(1);
+        expect(quads[0].equals(
+          new Quad(
+            new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')),
+            new NamedNode('p2'),
+            new NamedNode('o2')
+            )
+          )).toBe(true);
+      })
     );
 
     it('should decrease the size', () => {
-      expect(store.size).toEqual(5);
+      expect(store.size).toEqual(4);
     });
 
-    it('should match RDF* and normal quads at the same time', done => {
+    it('should match RDF-star and normal quads at the same time', done => {
       const stream = store.removeMatches(null, 'p1', 'o2');
       stream.on('end', () => {
-        expect(store.size).toEqual(3);
+        expect(store.size).toEqual(2);
         done();
       });
     });
 
     it('should allow matching using a quad', done => {
-      const stream = store.removeMatches(termToId(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1'))));
+      const stream = store.removeMatches(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')));
       stream.on('end', () => {
-        expect(store.size).toEqual(1);
+        expect(store.size).toEqual(0);
         done();
       });
     });
+
+    it(
+      'should allow matching using a quad and only match against relevant quads',
+      done => {
+        const s2 = new Store([
+          ...allQuads,
+          new Quad(
+            new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o2')),
+            new NamedNode('p1'),
+            new NamedNode('o2')),
+        ]);
+
+        const stream = s2.removeMatches(new Quad(new NamedNode('s1'), new NamedNode('p1'), new NamedNode('o1')));
+        stream.on('end', () => {
+          expect(s2.size).toEqual(2);
+          done();
+        });
+      }
+    );
+  });
+
+  // These tests should probably be broken in the future; they are here to serve to use that we should to a mver bump
+  // at such a time
```

Comment:
The portion following the semicolon needs rewriting. At present, it is indecipherable.